lib/tag-field.jsx

```diff
@@ -32,18 +55,24 @@ export default React.createClass( {
 		}
 	},
 
-	clearTextField: function() {
-		this.refs.tag.value = '';
-	},
-
 	addTag: function( tags ) {
-		tags = tags.trim().replace( /\s+/g, ',' ).split( ',' );
-		tags = this.props.tags.concat( tags );
-		this.props.onUpdateNoteTags( tags );
+		const newTags = tags.trim().replace( /\s+/g, ',' ).split( ',' );
+		this.props.onUpdateNoteTags( union( this.props.tags, newTags ) );
```

Comment:
it was concatenation that was providing duplicate tags. using a set union instead prevents that.